statsmodels/distributions/copula/copulas.py

```diff
@@ -177,6 +182,40 @@ def __init__(self, marginals, copula, copargs=()):
         self.copargs = copargs
         self.k_vars = len(marginals)
 
+    def random(self, n=1, random_state=None):
+        """Draw `n` in the half-open interval ``[0, 1)``.
+
+        Sample the joint distribution.
+
+        Parameters
+        ----------
+        n : int, optional
+            Number of samples to generate in the parameter space.
+            Default is 1.
+        random_state : {None, int, `numpy.random.Generator`}, optional
+            If `seed` is None the `numpy.random.Generator` singleton is used.
+            If `seed` is an int, a new ``Generator`` instance is used,
+            seeded with `seed`.
+            If `seed` is already a ``Generator`` instance then that instance is
+            used.
+
+        Returns
+        -------
+        sample : array_like (n, d)
+            Sample from the joint distribution.
+
+        """
+        rng = check_random_state(random_state)
+        if self.copula is None:
+            # this means marginals are independents
+            sample = rng.random((n, self.d))
+        else:
+            sample = self.copula.random(n, random_state=random_state)
```

Comment:
why does this not use `rng`?
what is return of check_random_state

Reply:
It's a typo and should read `sample = self.copula.random(n, random_state=rng)`. `check_random_state` returns an instance of `RandomState` or `Generator` (only if a generator was passed).